snapcraft/internal/build_providers/_multipass/_multipass.py

```diff
@@ -74,33 +114,37 @@ def provision_project(self, tarball: str) -> None:
 
         # First create a working directory
         self._multipass_cmd.execute(
-            command=["mkdir", self.project_dir], instance_name=self.instance_name
+            command=["mkdir", self._INSTANCE_PROJECT_DIR],
+            instance_name=self.instance_name,
         )
 
         # Then copy the tarball over
         destination = "{}:{}".format(self.instance_name, tarball)
         self._multipass_cmd.copy_files(source=tarball, destination=destination)
 
         # Finally extract it into project_dir.
-        extract_cmd = ["tar", "-xvf", tarball, "-C", self.project_dir]
+        extract_cmd = ["tar", "-xvf", tarball, "-C", self._INSTANCE_PROJECT_DIR]
         self._multipass_cmd.execute(
             command=extract_cmd, instance_name=self.instance_name
         )
 
+    def clean_project(self) -> bool:
+        was_cleaned = super().clean_project()
+        if was_cleaned:
+            self._multipass_cmd.delete(instance_name=self.instance_name)
+        return was_cleaned
+
     def build_project(self) -> None:
         # TODO add instance check.
-        # Use the full path as /snap/bin is not in PATH.
-        snapcraft_cmd = "cd {}; /snap/bin/snapcraft snap --output {}".format(
-            self.project_dir, self.snap_filename
-        )
         self._multipass_cmd.execute(
-            command=["sh", "-c", snapcraft_cmd], instance_name=self.instance_name
+            command=["snapcraft", "snap", "--output", self.snap_filename],
```

Comment:
Is `/snap/bin` on the PATH, now?

Reply:
yes,we could FQPath it though

Reply:
discard that, /snap/bin is not widespread across potential build environments.

Reply:
issue taken care of